lib/scope.js

```diff
@@ -48,10 +48,56 @@ function Scope(basePath, options) {
   this.port = null;
 
   if (!(basePath instanceof RegExp)) {
-    this.urlParts = url.parse(basePath);
-    this.port = this.urlParts.port || ((this.urlParts.protocol === 'http:') ? 80 : 443);
-    this.basePathname = this.urlParts.pathname.replace(/\/$/, '');
-    this.basePath = this.urlParts.protocol + '//' + this.urlParts.hostname + ':' + this.port;
+    this.urlPart = {};
+
+    // Extract protocol
+    var protocolDelim = new RegExp(/\/{2}/);
```

Comment:
Is there an advantage with using the `RegExp` constructor over just `var protocolDelim = /\/{2}/`?

Also what if someone does sth like `nock('http://example.com//foo')`? Maybe it would be better to split with `/:\/{2}/`? 

Could you add a comment explaining why we don’t use `url.parse()` here, maybe reference the pull request?